modelon/impact/client/jupyterhub/authorize.py

```diff
@@ -17,13 +18,27 @@ def _get_jupyter_token(credential_manager: CredentialManager, interactive: bool)
     return jupyter_token
 
 
+def _running_in_jupyterhub_environment() -> bool:
+    return os.environ.get("JUPYTERHUB_SERVICE_PREFIX") is not None
+
+
 def authorize(
     uri: URI,
     interactive: bool,
     context: Optional[Context] = None,
     credential_manager: Optional[CredentialManager] = None,
     service: Optional[sal.JupyterHubService] = None,
 ) -> Tuple[URI, sal.JupyterContext]:
+
+    jupyter_context = sal.JupyterContext(base=context)
+    service = service or sal.JupyterHubService()
+
+    if _running_in_jupyterhub_environment():
+        jupyter_context.token = os.environ.get('JUPYTERHUB_API_TOKEN', '')
```

Comment:
The problem that I see here is handling of connection from "other" jupyterhub (like on-prem -> impact.modelon.com). It seems the code will always pick-up "current" Jupyterhub environment and there's no obvious way to override that.